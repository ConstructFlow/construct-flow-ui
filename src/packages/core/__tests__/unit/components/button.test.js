const sass = require('sass');
const { colors } = require('../../shared/constants');

describe('Colors variables', () => {
  const result = sass
    .compile('./src/packages/core/src/components/button/_button.scss', { style: 'compressed' })
    .css.toString();

  it('Should match', () => {
    expect(result).toMatchSnapshot();
  });

  it('Should has full list of buttons', async () => {
    expect(result).toContain('.button');
    expect(result).toContain('.button--primary');
    expect(result).toContain('.button--danger');
    expect(result).toContain('.button--success');
    expect(result).toContain('.button--dark');
    expect(result).toContain('.button--brown');
    expect(result).toContain('.button--green');
    expect(result).toContain('.button--blue');
    expect(result).toContain('.button--turquoise');
    expect(result).toContain('.button--violet');
    expect(result).toContain('.button--yellow');
    expect(result).toContain('.button--orange');
    expect(result).toContain('.button--pink');
    expect(result).toContain('.button--red');
    expect(result).toContain('.button--loading');
  });

  it('Should has proper colors', async () => {
    expect(result).toContain(`.button--primary{background:${colors.colorBlue40};color:#fff}`);
    expect(result).toContain(`.button--danger{background:${colors.colorRed40};color:#fff}`);
    expect(result).toContain(`.button--success{background:${colors.colorGreen40};color:#fff}`);
    expect(result).toContain(`.button--dark{background:${colors.colorGray70};color:#fff}`);
    expect(result).toContain(`.button--gray{background:${colors.colorGray40};color:#fff}`);
    expect(result).toContain(`.button--brown{background:${colors.colorBrown50};color:#fff}`);
    expect(result).toContain(`.button--brown{background:${colors.colorBrown50};color:#fff}`);
    expect(result).toContain(`.button--green{background:${colors.colorGreen50};color:#fff}`);
    expect(result).toContain(`.button--blue{background:${colors.colorBlue50};color:#fff}`);
    expect(result).toContain(
      `.button--turquoise{background:${colors.colorTurquoise40};color:#fff}`
    );
    expect(result).toContain(`.button--violet{background:${colors.colorViolet40};color:#fff}`);
    expect(result).toContain(`.button--yellow{background:${colors.colorYellow60};color:#fff}`);
    expect(result).toContain(`.button--orange{background:${colors.colorOrange50};color:#fff}`);
    expect(result).toContain(`.button--pink{background:${colors.colorPink50};color:#fff}`);
    expect(result).toContain(`.button--red{background:${colors.colorRed50};color:#fff}`);
  });
});
