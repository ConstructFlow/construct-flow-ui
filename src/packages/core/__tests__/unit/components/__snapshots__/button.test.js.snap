// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors variables Should match 1`] = `"@-webkit-keyframes rotating{from{-webkit-transform:rotate(0deg);-o-transform:rotate(0deg);transform:rotate(0deg)}to{-webkit-transform:rotate(360deg);-o-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes rotating{from{-ms-transform:rotate(0deg);-moz-transform:rotate(0deg);-webkit-transform:rotate(0deg);-o-transform:rotate(0deg);transform:rotate(0deg)}to{-ms-transform:rotate(360deg);-moz-transform:rotate(360deg);-webkit-transform:rotate(360deg);-o-transform:rotate(360deg);transform:rotate(360deg)}}.button,.button--loading,.button--red,.button--pink,.button--orange,.button--yellow,.button--violet,.button--turquoise,.button--blue,.button--green,.button--brown,.button--gray,.button--dark,.button--success,.button--danger,.button--primary{cursor:pointer;display:inline-block;font-size:1rem;border:none;background:#ced4da;border-radius:.25rem;color:#495057;padding:.5rem 1rem;transition:background-color .2s linear;min-height:1.5rem;margin:0 0 .5rem 0}.button[disabled],[disabled].button--loading,[disabled].button--red,[disabled].button--pink,[disabled].button--orange,[disabled].button--yellow,[disabled].button--violet,[disabled].button--turquoise,[disabled].button--blue,[disabled].button--green,[disabled].button--brown,[disabled].button--gray,[disabled].button--dark,[disabled].button--success,[disabled].button--danger,[disabled].button--primary,.button:disabled,.button--loading:disabled,.button--red:disabled,.button--pink:disabled,.button--orange:disabled,.button--yellow:disabled,.button--violet:disabled,.button--turquoise:disabled,.button--blue:disabled,.button--green:disabled,.button--brown:disabled,.button--gray:disabled,.button--dark:disabled,.button--success:disabled,.button--danger:disabled,.button--primary:disabled{cursor:none;pointer-events:none;opacity:.7}.button:hover,.button--loading:hover,.button--red:hover,.button--pink:hover,.button--orange:hover,.button--yellow:hover,.button--violet:hover,.button--turquoise:hover,.button--blue:hover,.button--green:hover,.button--brown:hover,.button--gray:hover,.button--dark:hover,.button--success:hover,.button--danger:hover,.button--primary:hover{background:#adb5bd;color:#343a40}.button--primary{background:#42a5f5;color:#fff}.button--primary:hover{background:#2196f3;color:#fff}.button--danger{background:#ec4040;color:#fff}.button--danger:hover{background:#ef2b2b;color:#fff}.button--success{background:#2dc653;color:#fff}.button--success:hover{background:#25a244;color:#fff}.button--dark{background:#495057;color:#fff}.button--dark:hover{background:#343a40;color:#fff}.button--gray{background:#ced4da;color:#fff}.button--gray:hover{background:#adb5bd;color:#fff}.button--brown{background:#c38e70;color:#fff}.button--brown:hover{background:#b07d62;color:#fff}.button--green{background:#25a244;color:#fff}.button--green:hover{background:#208b3a;color:#fff}.button--blue{background:#2196f3;color:#fff}.button--blue:hover{background:#1e88e5;color:#fff}.button--turquoise{background:#00b4d8;color:#fff}.button--turquoise:hover{background:#0096c7;color:#fff}.button--violet{background:#b084d7;color:#fff}.button--violet:hover{background:#a674d2;color:#fff}.button--yellow{background:#ffe747;color:#fff}.button--yellow:hover{background:#ffdd1f;color:#fff}.button--orange{background:#ff7900;color:#fff}.button--orange:hover{background:#ff6d00;color:#fff}.button--pink{background:#ff85a1;color:#fff}.button--pink:hover{background:#ff7096;color:#fff}.button--red{background:#ef2b2b;color:#fff}.button--red:hover{background:#de1616;color:#fff}.button--loading{position:relative;cursor:default;text-shadow:none !important;color:rgba(0,0,0,0) !important;opacity:1;pointer-events:auto;transition:all 0s linear,opacity .1s ease}.button--loading:before{position:absolute;content:"";top:calc(50% - .625rem);left:calc(50% - .625rem);width:1rem;height:1rem;border-radius:500rem;border-top-left-radius:500rem;border-top-right-radius:500rem;border-bottom-right-radius:500rem;border-bottom-left-radius:500rem;border:.2em solid rgba(0,0,0,.15)}.button--loading:after{position:absolute;content:"";top:calc(50% - .625rem);left:calc(50% - .625rem);width:1rem;height:1rem;-webkit-animation:rotating .6s linear;animation:rotating .6s linear;-webkit-animation-iteration-count:infinite;animation-iteration-count:infinite;border-radius:500rem;border-top-left-radius:500rem;border-top-right-radius:500rem;border-bottom-right-radius:500rem;border-bottom-left-radius:500rem;border-color:#fff rgba(0,0,0,0) rgba(0,0,0,0);border-style:solid;border-width:.2em;box-shadow:0 0 0 1px rgba(0,0,0,0)}.button--mini{font-size:.75rem;line-height:1.5rem}.button--small{font-size:.9rem;line-height:1.8rem}.button--medium{font-size:1rem;line-height:2rem}.button--large{font-size:1.25rem;line-height:2.5rem}"`;
